src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -187,15 +190,15 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_GetSerialNumber(char * b
     CHIP_ERROR err;
 
     err = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
-#ifdef CHIP_DEVICE_CONFIG_USE_TEST_SERIAL_NUMBER
-    if (CHIP_DEVICE_CONFIG_USE_TEST_SERIAL_NUMBER[0] != 0 && err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+#ifdef CHIP_DEVICE_CONFIG_TEST_SERIAL_NUMBER
+    if (CHIP_DEVICE_CONFIG_TEST_SERIAL_NUMBER[0] != 0 && err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
```

Comment:
could we assume that CHIP_DEVICE_CONFIG_TEST_SERIAL_NUMBER is not empty if defined and save a conditional?

Reply:
well, we could but i'd rather be extra careful here

Reply:
FWIW, any reasonable compiler will turn `CHIP_DEVICE_CONFIG_TEST_SERIAL_NUMBER[0] != 0` into a constant expression.